code/modules/luck/luck.dm

```diff
@@ -0,0 +1,149 @@
+//Luckiness and unluckiness
+
+/mob/proc/luck()
+	var/luck = 0
+	if(base_luck)
+		luck += base_luck.base_luck()
+	//Adjust based on borne items
+	luck += borne_item_luckiness()
+	return luck
+
+/mob
+	var/datum/luckiness/base_luck
+
+/mob/proc/luck_adjust(var/luckchange, var/temporary = FALSE)
+	if(!base_luck)
+		base_luck = new /datum/luckiness
+	if(temporary)
+		base_luck.temporary_luckiness += luckchange
+	else
+		base_luck.permanent_luckiness += luckchange
+
+/mob/proc/borne_item_luckiness() //Check all inventory items for (un)luckiness.
+	var/total_item_luckiness = 0
+	var/list/equipped_items = get_equipped_items()
+	while(null in equipped_items)
+		equipped_items -= null
+	var/list/held_or_implanted_items = contents - equipped_items
+	//check hand slot items and implanted items
+	if(held_or_implanted_items.len)
+		for(var/i in 1 to held_or_implanted_items.len)
+			var/obj/item/thisitem = held_or_implanted_items[i]
+			if(thisitem.luckiness_validity & (LUCKINESS_WHEN_HELD | LUCKINESS_WHEN_HELD_RECURSIVE) && thisitem.luckiness)
+				total_item_luckiness += thisitem.luckiness
+			//check hand slot and implanted item contents recursively
+			if(thisitem.contents.len)
+				var/list/thisitemscontents = thisitem.contents
+				var/j = 1
+				while(j <= thisitemscontents.len)
+					var/obj/item/thiscontent = thisitemscontents[j]
+					if(thiscontent.contents.len)
+						thisitemscontents += thiscontent.contents
+					j += 1
```

Comment:
Should be a different type of loop entirely as I said above, but just out of curiosity, why is this one set up differently from the rest?

Reply:
It checks through all the items in `thisitemscontents` for contents, adding those contents to the end of `thisitemscontents`. So in that way you can get it to check the contents of the contents of the contents.. etc. Like a backpack containing an emergency survival box containing a first aid kit containing a clover. It stops when it runs out of items with contents. Since the length of `thisitemscontents` changes within the loop, I figured to use `while` instead of `for` since I thought `for` only checks the list at the beginning of the loop.

Reply:
This loop is exactly equivalent to an indexed for loop, but I didn't catch the list in question changing size, so what I said about changing all the loops to a for *in* loop doesn't apply to this one. As for `to`, I'm not actually sure if using a range like that works for this, but the *traditional* for loop syntax definitely does.
Side note: I don't remember if `contents` aliases. You should make sure this loop doesn't end up actually moving things from the inner objects into the outermost one.

Reply:
I changed the for loops to `in list` format. What do you mean by traditional for loop syntax?
So would either a while loop or an indexed for loop both work in this case? If so that's good to know.
>You should make sure this loop doesn't end up actually moving things from the inner objects into the outermost one.

What would be a good way to check that?

Reply:
By traditional, I mean the original one from C or whatever. Like (in BYOND) `for(var/i = 1; i <= thing.len; i++)`
A good way to check it is in-game. Just hold some nested items, do something that would call this, and check if they moved.

Reply:
Oh I see.
> A good way to check it is in-game. Just hold some nested items, do something that would call this, and check if they moved.

I tested and it turns out stuff does move around. Hmm.. I wonder how get the same functionality but avoid this.

Reply:
I added `.Copy()` and tested it by putting: 4 four-leaf clovers in: the pockets, the backpack, the emergency survival box in the backpack, and a first aid kid in the backpack. After playing scratch cards, nothing moved, and calling luck() by itself returned the proper value. So hopefully that's fixed.

Reply:
No, it's because `list2 = list1` means both variables refer to *the same list*, so you were adding the items in the contents of the inner items to the contents of the outer item.

Reply:
That makes sense I think.

Reply:
It's one of those weird things with BYOND. `contents` should probably be read-only with only `loc` being editable, but no, you can move things by adding them to something else's `contents` as well.